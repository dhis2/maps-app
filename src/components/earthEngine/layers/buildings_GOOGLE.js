import i18n from '@dhis2/d2-i18n'
import { EARTH_ENGINE_LAYER } from '../../../constants/layers.js'

export default {
    layer: EARTH_ENGINE_LAYER,
    format: 'FeatureCollection',
    layerId: 'GOOGLE/Research/open-buildings/v1/polygons',
    img: 'images/buildings.png',
    datasetId: 'GOOGLE/Research/open-buildings/v1/polygons',
    name: i18n.t('Building footprints'),
    description: i18n.t(
        'The outlines of buildings derived from high-resolution satellite imagery. Only for the continent of Africa.'
    ),
    notice: i18n.t(
        'Building counts are only available for smaller organisation unit areas.'
    ),
    error: i18n.t(
        'Select a smaller area or single organization unit to see the count of buildings.'
    ),
    source: 'NASA / USGS / JPL-Caltech / Google Earth Engine',
    sourceUrl: 'https://sites.research.google/open-buildings/',
    unit: i18n.t('Number of buildings'),
    aggregations: ['count'],
    defaultAggregations: ['count'],
    /* Not supported in UI yet
    customFilters: [
        {
            type: 'gt',
            arguments: ['area_in_meters', $1],
            valueType: 'number',
            label: 'Area in mÂ²',
        },
    ],
    */
    style: {
        color: '#FFA500',
        strokeWidth: 1,
    },
    opacity: 0.9,
}
