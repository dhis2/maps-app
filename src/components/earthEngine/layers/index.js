import buildings from './buildings_GOOGLE.js'
import elevation from './elevation_SRTM.js'
import landcover100m from './landcover_CGLS.js'
import landcover from './landcover_MCD12Q1.js'
import nighttimeLights from './legacy/nighttime_DMSP-OLS .js'
import precipitationLegacy from './legacy/precipitation_pentad_CHIRPS.js'
import temperatureLegacy from './legacy/temperature_MOD11A2v061.js'
import nitrogenDioxide from './nitrogen_dioxide_Sentinel-5P.js'
import ozone from './ozone_Sentinel-5P.js'
import particlePollution from './particle_pollution_CAMS.js'
import populationAgeSex from './population_age_sex_WorldPop.js'
import populationTotal from './population_total_WorldPop.js'
import precipitationDailyChirps from './precipitation_daily_CHIRPS.js'
import precipitationDaily from './precipitation_daily_ERA5-Land.js'
import precipitationMonthly from './precipitation_monthly_ERA5-Land.js'
import satelliteImagery from './satellite_imagery_Sentinel-2.js'
import sulfurDioxide from './sulfur_dioxide_Sentinel-5P.js'
import temperatureDaily from './temperature_daily_ERA5-Land.js'
import temperatureMax from './temperature_daily_max_ERA5-Land.js'
import temperatureMonthly from './temperature_monthly_ERA5-Land.js'

const earthEngineLayers = [
    buildings,
    elevation,
    landcover,
    landcover100m,
    nighttimeLights,
    nitrogenDioxide,
    ozone,
    particlePollution,
    populationTotal,
    populationAgeSex,
    precipitationDaily,
    precipitationDailyChirps,
    precipitationMonthly,
    precipitationLegacy,
    satelliteImagery,
    sulfurDioxide,
    temperatureDaily,
    temperatureMax,
    temperatureMonthly,
    temperatureLegacy,
]

export default earthEngineLayers
