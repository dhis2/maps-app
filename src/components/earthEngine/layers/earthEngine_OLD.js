import i18n from '@dhis2/d2-i18n'
import { EARTH_ENGINE_LAYER } from './layers.js'

export const earthEngineLayers = [
    {
        layerType: EARTH_ENGINE_LAYER,
        layerId: 'ECMWF/CAMS/NRT/NO2',
        datasetId: 'ECMWF/CAMS/NRT',
        format: 'ImageCollection',
        name: 'Nitrogen dioxide',
        description: 'Total column of nitrogen dioxide surface',
        source: 'Copernicus Climate Data Store / Google Earth Engine',
        unit: 'kg/m^2',
        // aggregations: ['min', 'max', 'mean', 'median', 'stdDev', 'variance'],
        // defaultAggregations: ['mean', 'min', 'max'],
        // periodType: 'daily',
        periodType: 'hourly',
        // periodReducer: 'sum',
        band: 'total_column_nitrogen_dioxide_surface',
        filters: [
            {
                type: 'eq',
                arguments: ['system:index', '$1'],
            },
        ],
        methods: [
            {
                name: 'multiply',
                arguments: [1000],
            },
        ],
        style: {
            min: 0,
            max: 0.05,
            palette: [
                '#fff7f3',
                '#fde0dd',
                '#fcc5c0',
                '#fa9fb5',
                '#f768a1',
                '#dd3497',
                '#ae017e',
                '#7a0177',
                '#49006a',
            ],
        },
        opacity: 0.9,
    },
    {
        layerType: EARTH_ENGINE_LAYER,
        layerId: 'ECMWF/CAMS/NRT/SO2',
        datasetId: 'ECMWF/CAMS/NRT',
        format: 'ImageCollection',
        name: 'Sulfur dioxide',
        description: 'Total column sulfur dioxide surface',
        source: 'Copernicus Climate Data Store / Google Earth Engine',
        unit: 'kg/m^2',
        // aggregations: ['min', 'max', 'mean', 'median', 'stdDev', 'variance'],
        // defaultAggregations: ['mean', 'min', 'max'],
        // periodType: 'daily',
        periodType: 'hourly',
        // periodReducer: 'sum',
        band: 'total_column_sulphur_dioxide_surface',
        filters: [
            {
                type: 'eq',
                arguments: ['system:index', '$1'],
            },
        ],
        methods: [
            {
                name: 'multiply',
                arguments: [1000],
            },
        ],
        style: {
            min: 0,
            max: 0.05,
            palette: [
                '#fff7f3',
                '#fde0dd',
                '#fcc5c0',
                '#fa9fb5',
                '#f768a1',
                '#dd3497',
                '#ae017e',
                '#7a0177',
                '#49006a',
            ],
        },
        opacity: 0.9,
    },
    {
        layerType: EARTH_ENGINE_LAYER,
        layerId: 'ECMWF/CAMS/NRT/CO',
        datasetId: 'ECMWF/CAMS/NRT',
        format: 'ImageCollection',
        name: 'Carbon monoxide',
        description: 'Total column carbon monoxide surface',
        source: 'Copernicus Climate Data Store / Google Earth Engine',
        unit: 'kg/m^2',
        // aggregations: ['min', 'max', 'mean', 'median', 'stdDev', 'variance'],
        // defaultAggregations: ['mean', 'min', 'max'],
        // periodType: 'daily',
        periodType: 'hourly',
        // periodReducer: 'sum',
        band: 'total_column_carbon_monoxide_surface',
        filters: [
            {
                type: 'eq',
                arguments: ['system:index', '$1'],
            },
        ],
        methods: [
            {
                name: 'multiply',
                arguments: [1000],
            },
        ],
        style: {
            min: 0,
            max: 3,
            palette: [
                '#fff7f3',
                '#fde0dd',
                '#fcc5c0',
                '#fa9fb5',
                '#f768a1',
                '#dd3497',
                '#ae017e',
                '#7a0177',
                '#49006a',
            ],
        },
        opacity: 0.9,
    },
    {
        layerType: EARTH_ENGINE_LAYER, // TODO: Remove?
        layerId: 'COPERNICUS/S2_SR_HARMONIZED',
        datasetId: 'COPERNICUS/S2_SR_HARMONIZED',
        format: 'ImageCollection',
        name: 'Sentinel-2 imagery',
        description: '',
        source: 'Copernicus / Google Earth Engine',
        periodType: 'range',
        periodReducer: 'median',
        cloudScore: {
            datasetId: 'GOOGLE/CLOUD_SCORE_PLUS/V1/S2_HARMONIZED',
            band: 'cs',
            clearTreshold: 0.6,
        },
        filters: [
            {
                type: 'date',
                // arguments: ['$1', '$2'],
                // arguments: ['2023-01-01', '2023-02-01'],
                // arguments: ['2023-06-01', '2023-07-01'],
                arguments: ['2022-08-27', '2022-09-05'], // Pakistan floods
                // arguments: ['2023-08-21', '2023-09-05'],
            },
        ],
        style: {
            bands: ['B4', 'B3', 'B2'], // red, green, blue
            min: 0,
            max: 2500,
        },
        opacity: 0.9,
    },
]

/*
// layerId should be unique
// datasetId is the Earth Engine dataset id
export const earthEngineLayers = () => [
    {
        layer: EARTH_ENGINE_LAYER,
        legacy: true, // Kept for backward compability
        layerId: 'WorldPop/GP/100m/pop',
        datasetId: 'WorldPop/GP/100m/pop',
        name: i18n.t('Population'),
        unit: i18n.t('people per hectare'),
        description: i18n.t('Estimated number of people living in an area.'),
        source: 'WorldPop / Google Earth Engine',
        sourceUrl:
            'https://developers.google.com/earth-engine/datasets/catalog/WorldPop_GP_100m_pop',
        img: 'images/population.png',
        defaultAggregations: ['sum', 'mean'],
        periodType: 'Yearly',
        filters: ({ id, name, year }) => [
            {
                id,
                name,
                type: 'eq',
                arguments: ['year', year],
            },
        ],
        mosaic: true,
        params: {
            min: 0,
            max: 10,
            palette: [
                '#fee5d9',
                '#fcbba1',
                '#fc9272',
                '#fb6a4a',
                '#de2d26',
                '#a50f15',
            ], // Reds
        },
        opacity: 0.9,
    },
    {
        layer: EARTH_ENGINE_LAYER,
        legacy: true, // Kept for backward compability
        layerId: 'WorldPop/POP',
        datasetId: 'WorldPop/POP',
        name: i18n.t('Population'),
        unit: i18n.t('people per kmÂ²'),
        description: i18n.t('Estimated number of people living in an area.'),
        source: 'WorldPop / Google Earth Engine',
        sourceUrl:
            'https://explorer.earthengine.google.com/#detail/WorldPop%2FPOP',
        img: 'images/population.png',
        periodType: 'Yearly',
        filters: ({ id, name, year }) => [
            {
                id,
                name,
                type: 'eq',
                arguments: ['year', year],
            },
            {
                type: 'eq',
                arguments: ['UNadj', 'yes'],
            },
        ],
        mosaic: true,
        params: {
            min: 0,
            max: 1000,
            palette: [
                '#fee5d9',
                '#fcbba1',
                '#fc9272',
                '#fb6a4a',
                '#de2d26',
                '#a50f15',
            ], // Reds
        },
        methods: {
            multiply: [100], // Convert from people/hectare to people/km2
        },
        opacity: 0.9,
    },
]
*/

export const getEarthEngineLayer = (id) =>
    earthEngineLayers.find((l) => l.layerId === id)
