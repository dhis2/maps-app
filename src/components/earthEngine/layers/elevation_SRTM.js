import i18n from '@dhis2/d2-i18n'
import { EARTH_ENGINE_LAYER } from '../../../constants/layers.js'

export default {
    layerType: EARTH_ENGINE_LAYER,
    layerId: 'USGS/SRTMGL1_003',
    datasetId: 'USGS/SRTMGL1_003',
    format: 'Image',
    img: 'images/elevation.png',
    name: i18n.t('Elevation'),
    unit: i18n.t('meters'),
    description: i18n.t('Elevation above sea-level.'),
    source: 'NASA / USGS / JPL-Caltech / Google Earth Engine',
    sourceUrl:
        'https://explorer.earthengine.google.com/#detail/USGS%2FSRTMGL1_003',
    aggregations: ['min', 'max', 'mean', 'median', 'stdDev', 'variance'],
    defaultAggregations: ['mean', 'min', 'max'],
    band: 'elevation',
    style: {
        min: 0,
        max: 1500,
        palette: [
            '#ffffd4',
            '#fee391',
            '#fec44f',
            '#fe9929',
            '#d95f0e',
            '#993404',
        ], // YlOrBr
    },
    maskOperator: 'gte',
    opacity: 0.9,
}
