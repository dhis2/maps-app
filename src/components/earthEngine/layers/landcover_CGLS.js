import i18n from '@dhis2/d2-i18n'
import { EARTH_ENGINE_LAYER } from '../../../constants/layers.js'

export default {
    layerType: EARTH_ENGINE_LAYER,
    format: 'ImageCollection',
    layerId: 'COPERNICUS/Landcover/100m/Proba-V-C3/Global',
    img: 'images/landcover-copernicus.jpeg',
    datasetId: 'COPERNICUS/Landcover/100m/Proba-V-C3/Global',
    name: i18n.t('Landcover Copernicus'),
    description: i18n.t('Distinct landcover types collected from satellites.'),
    source: 'Copernicus / Google Earth Engine',
    periodType: 'YEARLY',
    band: 'discrete_classification',
    defaultAggregations: 'percentage',
    popup: '{name}: {value}', // In use?
    filters: [
        {
            type: 'eq',
            arguments: ['system:index', '$1'],
        },
    ],
    style: [
        {
            value: 0,
            name: i18n.t('Unknown'),
            color: '#282828',
        },
        {
            value: 20,
            name: i18n.t('Shrubs'),
            color: '#ffbb22',
        },
        {
            value: 30,
            name: i18n.t('Herbaceous vegetation'),
            color: '#ffff4c',
        },
        {
            value: 40,
            name: i18n.t('Agriculture'),
            color: '#f096ff',
        },
        {
            value: 50,
            name: i18n.t('Urban / built up'),
            color: '#fa0000',
        },
        {
            value: 60,
            name: i18n.t('Bare / sparse vegetation'),
            color: '#b4b4b4',
        },
        {
            value: 70,
            name: i18n.t('Snow and ice'),
            color: '#f0f0f0',
        },
        {
            value: 80,
            name: i18n.t('Permanent water bodies'),
            color: '#0032c8',
        },
        {
            value: 90,
            name: i18n.t('Herbaceous wetland'),
            color: '#0096a0',
        },
        {
            value: 100,
            name: i18n.t('Moss and lichen'),
            color: '#fae6a0',
        },
        {
            value: 111,
            name: i18n.t('Closed forest, evergreen needle leaf'),
            color: '#58481f',
        },
        {
            value: 112,
            name: i18n.t('Closed forest, evergreen broad leaf'),
            color: '#009900',
        },
        {
            value: 113,
            name: i18n.t('Closed forest, deciduous needle leaf'),
            color: '#70663e',
        },
        {
            value: 114,
            name: i18n.t('Closed forest, deciduous broad leaf'),
            color: '#00cc00',
        },
        {
            value: 115,
            name: i18n.t('Closed forest, mixed'),
            color: '#4e751f',
        },
        {
            value: 116,
            name: i18n.t('Closed forest, other'),
            color: '#007800',
        },
        {
            value: 121,
            name: i18n.t('Open forest, evergreen needle leaf'),
            color: '#666000',
        },
        {
            value: 122,
            name: i18n.t('Open forest, evergreen broad leaf'),
            color: '#8db400',
        },
        {
            value: 123,
            name: i18n.t('Open forest, deciduous needle leaf'),
            color: '#8d7400',
        },
        {
            value: 124,
            name: i18n.t('Open forest, deciduous broad leaf'),
            color: '#a0dc00',
        },
        {
            value: 125,
            name: i18n.t('Open forest, mixed'),
            color: '#929900',
        },
        {
            value: 126,
            name: i18n.t('Open forest, other'),
            color: '#648c00',
        },
        {
            value: 200,
            name: i18n.t('Water'),
            color: '#000080',
        },
    ],
}
