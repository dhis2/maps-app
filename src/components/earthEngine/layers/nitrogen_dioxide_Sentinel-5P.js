import i18n from '@dhis2/d2-i18n'
import { EARTH_ENGINE_LAYER } from '../../../constants/layers.js'

export default {
    layerType: EARTH_ENGINE_LAYER,
    format: 'ImageCollection',
    layerId: 'COPERNICUS/S5P/NRTI/L3_NO2',
    datasetId: 'COPERNICUS/S5P/NRTI/L3_NO2',
    img: 'images/nitrogen-dioxide.png',
    name: i18n.t('Nitrogen dioxide (NO2)'),
    description:
        'Total vertical column of Nitrogen dioxide (NO2). This gas enters the atmosphere as a result of human activities (notably fossil fuel combustion and biomass burning) and natural processes (wildfires, lightning, and microbiological processes in soils).',
    source: 'European Union / ESA / Copernicus / Google Earth Engine',
    sourceUrl:
        'https://developers.google.com/earth-engine/datasets/catalog/COPERNICUS_S5P_NRTI_L3_NO2',
    unit: 'mol/m^2',
    periodType: 'WEEKLY',
    periodReducer: 'mean',
    band: 'NO2_column_number_density',
    filters: [
        {
            type: 'date',
            arguments: ['$1', '$2'],
        },
    ],
    style: {
        min: 0,
        max: 0.0003,
        palette: [
            '#feebe2',
            '#fcc5c0',
            '#fa9fb5',
            '#f768a1',
            '#dd3497',
            '#ae017e',
            '#7a0177',
        ],
    },
    opacity: 0.8,
}
