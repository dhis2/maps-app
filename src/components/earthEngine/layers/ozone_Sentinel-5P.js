import i18n from '@dhis2/d2-i18n'
import { EARTH_ENGINE_LAYER } from '../../../constants/layers.js'

export default {
    layerType: EARTH_ENGINE_LAYER,
    format: 'ImageCollection',
    layerId: 'COPERNICUS/S5P/NRTI/L3_O3',
    datasetId: 'COPERNICUS/S5P/NRTI/L3_O3',
    img: 'images/ozone.png',
    name: i18n.t('Ozone'),
    description:
        'Total atmospheric column of O3 between the surface and the top of atmosphere,',
    source: 'European Union / ESA / Copernicus / Google Earth Engine',
    sourceUrl:
        'https://developers.google.com/earth-engine/datasets/catalog/COPERNICUS_S5P_NRTI_L3_O3',
    unit: 'mol/m^2',
    periodType: 'WEEKLY',
    periodReducer: 'mean',
    band: 'O3_column_number_density',
    filters: [
        {
            type: 'date',
            arguments: ['$1', '$2'],
        },
    ],
    style: {
        min: 0,
        max: 0.2,
        palette: [
            '#fff7f3',
            '#fde0dd',
            '#fcc5c0',
            '#fa9fb5',
            '#f768a1',
            '#dd3497',
            '#ae017e',
            '#7a0177',
        ],
    },
    opacity: 0.8,
}
