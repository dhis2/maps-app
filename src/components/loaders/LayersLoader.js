import React, { useEffect } from 'react'
import { useSelector, useDispatch } from 'react-redux'
import { setLayerLoading, updateLayer } from '../../actions/layers.js'
import { EVENT_LAYER } from '../../constants/layers.js'
import LayerLoader from './LayerLoader.js'

const LayersLoader = () => {
    const dataTable = useSelector((state) => state.dataTable)
    const layersToLoad = useSelector(({ map }) =>
        map.mapViews.filter(
            ({
                id,
                isLoading,
                isLoaded,
                isExtended,
                layer: layerType,
                serverCluster,
            }) => {
                // The layer is currently being loaded - don't load again
                if (isLoading) {
                    return false
                } else {
                    // The layer is not loaded - load it
                    if (!isLoaded) {
                        return true
                    }

                    // The layer is loaded but the data table is now displayed and
                    // event extended data hasn't been loaded yet - so load it
                    if (
                        layerType === EVENT_LAYER &&
                        id === dataTable &&
                        !isExtended &&
                        !serverCluster
                    ) {
                        return true
                    }

                    return false
                }
            }
        )
    )
    const dispatch = useDispatch()

    const onLoad = (layer) => dispatch(updateLayer(layer))

    useEffect(() => {
        layersToLoad.forEach((layer) => dispatch(setLayerLoading(layer.id)))
    }, [layersToLoad, dispatch])

    if (!layersToLoad.length) {
        return null
    }

    return layersToLoad.map((config) => (
        <LayerLoader
            key={config.id}
            config={config}
            onLoad={onLoad}
            dataTableOpen={!!dataTable}
        />
    ))
}

export default LayersLoader
