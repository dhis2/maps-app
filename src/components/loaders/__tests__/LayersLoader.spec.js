import { render } from '@testing-library/react'
import React from 'react'
import { Provider } from 'react-redux'
import configureMockStore from 'redux-mock-store'
import { EVENT_LAYER } from '../../../constants/layers.js'
import LayersLoader from '../LayersLoader.js'

const mockStore = configureMockStore()

const id = 'thelayerid'

jest.mock(
    '../LayerLoader.js',
    () =>
        function MockLayerLoader() {
            return <div>LayerLoader</div>
        }
)

const testCases = [
    {
        testTitle: 'renders 0 layerloaders if no layers',
        store: {
            map: {
                mapViews: [],
            },
            dataTable: null,
        },
    },
    {
        testTitle: 'renders 0 layerloaders if currently loading the layer',
        store: {
            map: {
                mapViews: [{ isLoading: true, isLoaded: false }],
            },
            dataTable: null,
        },
    },
    {
        testTitle:
            'renders 0 layerloaders if currently loading regardless of whether loaded already',
        store: {
            map: {
                mapViews: [
                    {
                        isLoading: true,
                        isLoaded: true,
                    },
                ],
            },
            dataTable: null,
        },
    },
    {
        testTitle:
            'renders 1 layerloader if not currently loading and not loaded',
        store: {
            map: {
                mapViews: [
                    {
                        isLoading: false,
                        isLoaded: false,
                    },
                ],
            },
            dataTable: null,
        },
    },
    {
        testTitle:
            'renders 1 layerloader if not currently loading and is loaded but need extended data',
        store: {
            map: {
                mapViews: [
                    {
                        id,
                        layer: EVENT_LAYER,
                        isLoading: false,
                        isLoaded: true,
                        isExtended: false,
                        serverCluster: false,
                    },
                ],
            },
            dataTable: id,
        },
    },
    {
        testTitle:
            'renders 0 layerloaders if not currently loading and loaded and extended data also loaded',
        store: {
            map: {
                mapViews: [
                    {
                        id,
                        layer: EVENT_LAYER,
                        isLoading: false,
                        isLoaded: true,
                        isExtended: true,
                        serverCluster: false,
                    },
                ],
            },
            dataTable: id,
        },
    },
]

describe('LayersLoader', () => {
    testCases.forEach(({ testTitle, store }) => {
        it(testTitle, () => {
            const { container } = render(
                <Provider store={mockStore(store)}>
                    <LayersLoader />
                </Provider>
            )
            expect(container).toMatchSnapshot()
        })
    })
})
