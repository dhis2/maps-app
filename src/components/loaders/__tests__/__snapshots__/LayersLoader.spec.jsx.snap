// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LayersLoader renders 0 layerloaders if currently loading regardless of whether loaded already 1`] = `<div />`;

exports[`LayersLoader renders 0 layerloaders if currently loading the layer 1`] = `<div />`;

exports[`LayersLoader renders 0 layerloaders if no layers 1`] = `<div />`;

exports[`LayersLoader renders 0 layerloaders if not currently loading and loaded and extended data also loaded 1`] = `<div />`;

exports[`LayersLoader renders 1 layerloader if not currently loading and is loaded but need extended data 1`] = `
<div>
  <div>
    LayerLoader
  </div>
</div>
`;

exports[`LayersLoader renders 1 layerloader if not currently loading and not loaded 1`] = `
<div>
  <div>
    LayerLoader
  </div>
</div>
`;
